diff -ur wvdial-1.54.0/wvdial.1 wvdial-1.54.0.cfgstdin/wvdial.1
--- wvdial-1.54.0/wvdial.1	2003-11-14 20:46:13.000000000 +0100
+++ wvdial-1.54.0.cfgstdin/wvdial.1	2006-07-18 17:55:09.000000000 +0200
@@ -65,6 +65,11 @@
 configurations, or you want to avoid having dial-up information (usernames,
 passwords, calling card numbers, etc.) in a system wide configuration file.
 .TP
+.I "\-\-config-stdin"
+Have wvdial read its configuration from stdin (instead of /etc/wvdial.conf).
+This is mainly useful if you do not want to put the configuration data to a file
+if you call wvdial from within another program context.
+.TP
 .I "\-\-no-syslog"
 Don't output debug information to the syslog daemon (only useful together
 with --chat).
diff -ur wvdial-1.54.0/wvdial.cc wvdial-1.54.0.cfgstdin/wvdial.cc
--- wvdial-1.54.0/wvdial.cc	2003-11-14 20:46:13.000000000 +0100
+++ wvdial-1.54.0.cfgstdin/wvdial.cc	2006-07-26 13:09:39.000000000 +0200
@@ -16,6 +16,8 @@
 
 #include <signal.h>
 #include <sys/wait.h>
+#include <sys/time.h>
+#include <sys/types.h>
 #include <unistd.h>
 
 volatile bool want_to_die = false;
@@ -119,6 +121,30 @@
 		    return 1;			    
 		}
 	    }
+#ifndef _WIN32
+	    if( !strcmp( argv[i], "--config-stdin" ) ) 
+	    {
+	        // wait up to 10 seconds for data
+	      fd_set rfds;
+	      struct timeval tv;
+	      int retval = 1;
+	      FD_ZERO(&rfds);
+	      FD_SET(STDIN_FILENO, &rfds);
+	      tv.tv_sec  = 10;
+	      tv.tv_usec =  0;
+	      retval = select( STDIN_FILENO + 1, &rfds, NULL, NULL, &tv);
+	      if ( retval > 0 ) {
+		// data has arrived
+	        haveconfig = 1;
+	        cfg.load_file( STDIN_FILENO );
+	        continue;
+	      } else {
+		// no config data
+		log("no config data could be read from stdin");
+		exit( 3 );
+	      }
+	    }
+#endif
             if( strchr( argv[i], '=' ) ) {
                 havecmdlineopts = 1;
                 cmdlineopts.append(new WvString(argv[i]),true);
Nur in wvdial-1.54.0.cfgstdin/: wvdial.cc~.
diff -ur wvdial-1.54.0/wvdialtext.cc wvdial-1.54.0.cfgstdin/wvdialtext.cc
--- wvdial-1.54.0/wvdialtext.cc	2003-11-14 20:46:13.000000000 +0100
+++ wvdial-1.54.0.cfgstdin/wvdialtext.cc	2006-07-18 18:16:09.000000000 +0200
@@ -18,6 +18,7 @@
 "\n"
 "  options:  --config configfile    use configfile instead of /etc/wvdial.conf\n"
 " 				    or $HOME/.wvdial.conf\n"
+"	     --config-stdin         read config from stdin\n"
 "	     --chat		    used when running wvdial from pppd\n"
 "  	     --help		    display this help and exit\n"
 " 	     --version		    output version information and exit\n"
Nur in wvdial-1.54.0.cfgstdin/: wvdialtext.cc~.
