diff -urN wvdial-1.54.0/wvdialer.cc wvdial-1.54.0.tn/wvdialer.cc
--- wvdial-1.54.0/wvdialer.cc	2003-11-14 20:46:13.000000000 +0100
+++ wvdial-1.54.0.tn/wvdialer.cc	2006-07-18 12:20:02.000000000 +0200
@@ -601,6 +601,15 @@
     	{ "Password Prompt", &options.pass_prompt,  NULL, "",               0 },
     	{ "PPPD Path",       &options.where_pppd,   NULL, "/usr/sbin/pppd", 0 },
         { "PPPD Option",     &options.pppd_option,  NULL, "",		    0 },
+        { "PPPD Option 1",   &options.pppd_option1, NULL, "",		    0 },
+        { "PPPD Option 2",   &options.pppd_option2, NULL, "",		    0 },
+        { "PPPD Option 3",   &options.pppd_option3, NULL, "",		    0 },
+        { "PPPD Option 4",   &options.pppd_option4, NULL, "",		    0 },
+        { "PPPD Option 5",   &options.pppd_option5, NULL, "",		    0 },
+        { "PPPD Option 6",   &options.pppd_option6, NULL, "",		    0 },
+        { "PPPD Option 7",   &options.pppd_option7, NULL, "",		    0 },
+        { "PPPD Option 8",   &options.pppd_option8, NULL, "",		    0 },
+        { "PPPD Option 9",   &options.pppd_option9, NULL, "",		    0 },
     	{ "Force Address",   &options.force_addr,   NULL, "",               0 },
     	{ "Remote Name",     &options.remote,       NULL, "*",              0 },
     	{ "Default Reply",   &options.default_reply,NULL, "ppp",	    0 },
@@ -1205,6 +1214,15 @@
 	options.new_pppd && options.auto_dns ? "usepeerdns"	   : NULL,
 	options.new_pppd && options.isdn     ? "default-asyncmap"  : NULL,
 	options.new_pppd && options.pppd_option[0] ? (const char *) options.pppd_option : NULL,
+	options.new_pppd && options.pppd_option1[0] ? (const char *) options.pppd_option1 : NULL,
+	options.new_pppd && options.pppd_option2[0] ? (const char *) options.pppd_option2 : NULL,
+	options.new_pppd && options.pppd_option3[0] ? (const char *) options.pppd_option3 : NULL,
+	options.new_pppd && options.pppd_option4[0] ? (const char *) options.pppd_option4 : NULL,
+	options.new_pppd && options.pppd_option5[0] ? (const char *) options.pppd_option5 : NULL,
+	options.new_pppd && options.pppd_option6[0] ? (const char *) options.pppd_option6 : NULL,
+	options.new_pppd && options.pppd_option7[0] ? (const char *) options.pppd_option7 : NULL,
+	options.new_pppd && options.pppd_option8[0] ? (const char *) options.pppd_option8 : NULL,
+	options.new_pppd && options.pppd_option9[0] ? (const char *) options.pppd_option9 : NULL,
 	options.new_pppd && options.idle_seconds >= 0 ? "idle"	   : NULL, 
 	options.new_pppd && options.idle_seconds >= 0 ? (const char *)idle_seconds : NULL, 
 	"logfd", buffer1,
diff -urN wvdial-1.54.0/wvdialer.h wvdial-1.54.0.tn/wvdialer.h
--- wvdial-1.54.0/wvdialer.h	2003-11-14 20:46:13.000000000 +0100
+++ wvdial-1.54.0.tn/wvdialer.h	2006-07-18 12:20:58.000000000 +0200
@@ -112,6 +112,15 @@
 	WvString	        pass_prompt;
 	WvString	        where_pppd;
 	WvString	        pppd_option;
+	WvString	        pppd_option1;
+	WvString	        pppd_option2;
+	WvString	        pppd_option3;
+	WvString	        pppd_option4;
+	WvString	        pppd_option5;
+	WvString	        pppd_option6;
+	WvString	        pppd_option7;
+	WvString	        pppd_option8;
+	WvString	        pppd_option9;
 	WvString	        force_addr;
 	WvString	        remote;
 	WvString	        default_reply;
